path/path.go

```diff
@@ -36,3 +43,42 @@ func (p Path) Segments() []string {
 func (p Path) String() string {
 	return string(p)
 }
+
+func ParsePath(txt string) (Path, error) {
+	kp, err := ParseKeyToPath(txt)
+	if err == nil {
+		return kp, nil
+	}
+	parts := strings.Split(txt, "/")
+	if len(parts) < 3 {
+		return "", ErrBadPath
+	}
```

Comment:
may want to do the `parts := strings.Split(txt, "/")` part first, as it's faster than b58 decoding. maybe:

```go
parts := strings.Split(txt, "/")
if len(parts) == 1 {
  kp, err := ParseKeyToPath(txt)
  if err == nil {
    return kp, nil
  }
}
if len(parts) < 3 {
  return "", ErrBadPath
}
```